src/analysis/processing/qgsalgorithmstdbscanclustering.cpp

```diff
@@ -0,0 +1,340 @@
+
+void QgsStDbscanClusteringAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ),
+                QObject::tr( "Input layer" ), QList< int >() << QgsProcessing::TypeVectorPoint ) );
+  addParameter( new QgsProcessingParameterField( QStringLiteral( "DATETIME_FIELD" ),
+                QObject::tr( "Date/time field" ), QVariant(),
+                QStringLiteral( "INPUT" ), QgsProcessingParameterField::DateTime, false, false ) );
+
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "MIN_SIZE" ), QObject::tr( "Minimum cluster size" ),
+                QgsProcessingParameterNumber::Integer, 5, false, 1 ) );
+  addParameter( new QgsProcessingParameterDistance( QStringLiteral( "EPS1" ),
+                QObject::tr( "Maximum distance between clustered points" ), 1, QStringLiteral( "INPUT" ), false, 0 ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "EPS2" ),
+                QObject::tr( "Maximum date/time interval (in days unit) between clustered points" ), QgsProcessingParameterNumber::Double, 1, false, 0 ) );
+
+  auto dbscanStarParam = std::make_unique<QgsProcessingParameterBoolean>( QStringLiteral( "DBSCAN*" ),
+                         QObject::tr( "Treat border points as noise (DBSCAN*)" ), false, true );
+  dbscanStarParam->setFlags( dbscanStarParam->flags() | QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( dbscanStarParam.release() );
+
+  auto fieldNameParam = std::make_unique<QgsProcessingParameterString>( QStringLiteral( "FIELD_NAME" ),
+                        QObject::tr( "Cluster field name" ), QStringLiteral( "CLUSTER_ID" ) );
+  fieldNameParam->setFlags( fieldNameParam->flags() | QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( fieldNameParam.release() );
+  auto sizeFieldNameParam = std::make_unique<QgsProcessingParameterString>( QStringLiteral( "SIZE_FIELD_NAME" ),
+                            QObject::tr( "Cluster size field name" ), QStringLiteral( "CLUSTER_SIZE" ) );
+  sizeFieldNameParam->setFlags( sizeFieldNameParam->flags() | QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( sizeFieldNameParam.release() );
+
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ), QObject::tr( "Clusters" ), QgsProcessing::TypeVectorPoint ) );
+
+  addOutput( new QgsProcessingOutputNumber( QStringLiteral( "NUM_CLUSTERS" ), QObject::tr( "Number of clusters" ) ) );
+}
+
+QString QgsStDbscanClusteringAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Clusters point features based on a 2D implementation of Density-based spatiotemporal clustering of applications with noise (ST-DBSCAN) algorithm.\n\n"
+                      "The algorithm requires three parameters, a minimum cluster size (“minPts”), the maximum distance allowed between clustered points (“eps1”), and the maximum datetime interval allowed between cluster points (“eps2”)." );
```

Comment:
Is this necessary? It simply mentions what is shown in the GUI. I'd instead expect explanation on what/how the "Density-based spatiotemporal clustering ..." is/works.
Also, it leads to the question: does it mean that the date/time field is/should be (tagged) optional?

Reply:
@DelazJ , the help string matches what we have for the DBSCAN clustering algorithm.

And no, the date/time field is _not_ optional, it'd defeat the whole purpose :) for a spatial-only clustering, DBSCAN is the algorithm to use.

Reply:
> the help string matches what we have for the DBSCAN clustering algorithm.

Maybe the DBSCAN help as is is not the best template...?
> And no, the date/time field is not optional, it'd defeat the whole purpose :)

Agreed 😄. What I meant is: it's stated **`requires three parameters`** and there are 4 in the GUI (with none optional)

Reply:
I'll just go ahead and remove the paragraph then.